### PR TITLE
Workflow horizontal extension

### PR DESCRIPTION
**Pull request by laoneo**
Should illustrate a basic horizontal extension which adds state information to articles.

### CODE REVIEW

File: extensions/Workflow/entities/StateEntity.xml [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+            <validation rule="integer"/>
+        </field>
+    </fields>
+
+    <relations>
+    	<belongsTo name="entity_id"
+                   entity="Article"
+                   label="JFIELD_PARENT_LABEL"
+                   description="JFIELD_PARENT_DESC"
+        />
```
Comment (nibra) [inline extensions/Workflow/entities/StateEntity.xml:25]: Either adopt label and description, or just omit them.

File: extensions/Workflow/data/states.csv [hunk 2]
```diff
@@ -0,0 +1,6 @@
+id,title
+1,"Published"
+2,"Unpublished"
+3,"Trashed"
+4,"Archived"
+5,"Under Review"
```
Comment (nibra) [inline extensions/Workflow/data/states.csv]: Remember newline at the end of the file.

File: extensions/Workflow/ContentType/StateEntity.php [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+
+		$text = 'No State!';
+
+		if (isset($item->stateEntities) && $rel = $item->stateEntities->getAll())
+		{
+			$text = $rel[0]->state->title;
+		}
+		parent::__construct('State: ' . $item->stateEntities->getAll()[0]->state->title);
+	}
+}
```
Comment (nibra) [inline extensions/Workflow/ContentType/StateEntity.php:43]: Ok for now, but should be implemented as a callback for the renderer in the end.

File: extensions/UpperCase/config/extension.yml [hunk 4]
```diff
@@ -1,5 +0,0 @@
-listeners:
-    UpperCaseListener:
-        class: \Joomla\Extension\UpperCase\Listener\UpperCaseListener
-        events:
-            onBeforeRenderCompound: toUpperCase
```
Comment (nibra) [inline extensions/UpperCase/config/extension.yml:5]: Please also remove the test.

File: extensions/Workflow/entities/StateEntity.xml [hunk 5]
```diff
@@ -0,0 +1,35 @@
...
+    	<belongsTo name="state_id"
+                   entity="State"
+                   label="JFIELD_PARENT_LABEL"
+                   description="JFIELD_PARENT_DESC"
```
Comment (nibra) [inline extensions/Workflow/entities/StateEntity.xml]: Also here: Just omit label and description.

**Review by nibra — CHANGES_REQUESTED** [review]
Good starting point!

**Review by laoneo — COMMENTED** [review]
Labels are omitted and test removed

**Review by nibra — APPROVED** [review]

**Pull request merged**